### ISSUE TITLE
HTTP Patcher assumes options argument is an object

### ISSUE THREAD

**Issue by omsmith** [aws/aws-xray-sdk-node#16]
[node.js HTTP documentation](https://nodejs.org/api/http.html#http_http_request_options_callback) shows that `http.request` (or `http.get`, but there [really isn't a difference](https://github.com/nodejs/node/blob/5164a12618cf60150db621aebe1f3d89efb9b9aa/lib/http.js#L41)) can accept an `object`, `string` or somewhat more recently, an `URL` instance.

For the latter two, they are converted to the regular option object, and `GET` is assumed.

> options can be an object, a string, or a URL object. If options is a string, it is automatically parsed with url.parse(). If it is a URL object, it will be automatically converted to an ordinary options object.

HTTP implementation showing the parsing: https://github.com/nodejs/node/blob/5164a12618cf60150db621aebe1f3d89efb9b9aa/lib/_http_client.js#L57

Currently, making an outgoing http request with a string argument results in this error.

```
TypeError: Cannot set property 'X-Amzn-Trace-Id' of undefined
    at captureOutgoingHTTPs (/home/ec2-user/examples/1/node_modules/aws-xray-sdk-core/lib/patchers/http_p.js:88:40)
    at Object.captureHTTPsGet [as get] (/home/ec2-user/examples/1/node_modules/aws-xray-sdk-core/lib/patchers/http_p.js:148:12)
```

**Comment by haotianw465** [comment]
I found the same problem during testing. Will fix it soon. Thanks.

**Comment by haotianw465** [comment]
This fix is merged in and will go out with the next release.

**Comment by haotianw465** [comment]
Fix rolled out with 1.2.0 release.
